### PR TITLE
Move more of the videosource code to the videosourcewrappers file

### PR DESCRIPTION
**Pull request by randhid**
This pr simply moves code around so that the structs implementing videosources are all in one file.

### CODE REVIEW

**Review by hexbabe — COMMENTED** [review]
No weird build flag interference here?

**Comment by randhid** [comment]
> No weird build flag interference here?

Not according to my tests, but we'll see what CI says.

File: components/camera/videosourcewrappers.go [hunk 1]
```diff
@@ -36,6 +36,66 @@ func FromVideoSource(name resource.Name, src VideoSource, logger logging.Logger)
...
 
+type sourceBasedCamera struct {
+	resource.Named
+	resource.AlwaysRebuild
+	VideoSource
+	rtpPassthroughSource rtppassthrough.Source
+	logging.Logger
+}
+
+func (vs *sourceBasedCamera) SubscribeRTP(
```
Comment (seanavery) [inline components/camera/videosourcewrappers.go:47]: [q] Trying to figure out why we define `SubscribeRTP` and `Unsubscribe` for both `sourceBasedCamera`  and `videoSource`.

**Review by seanavery — APPROVED** [review]

File: components/camera/videosourcewrappers.go (hunk 1 above)
Comment (randhid) [reply components/camera/videosourcewrappers.go:47]: They're both implementing a VideoSource go interface interface/class, which can optionally have a `SubscribeRTP` method. There are other implementations of the `VideoSource` interface, for example `filesource` which is used in `image_file`. All of these structs implement the Camera/VideoSource methods, and can optionally implement `Subscribe/UnsubscribeRTP` to allow the RTP packet pass-through that @nicksanford wrote to be used by a camera's implementation.

Comment (randhid) [reply components/camera/videosourcewrappers.go:47]: This is also why in the constructors for different go cameras, we use the helper functions in this file to make one of the video sources that implements all of these methods. It's an example of trying ot use single point of truth/dry.

**Pull request merged**